pkg/crc/cluster/cluster.go

```diff
@@ -138,6 +139,41 @@ func StopAndRemovePodsInVM(sshRunner *ssh.SSHRunner) error {
 	return errors.RetryAfter(2, stopAndRemovePods, 2*time.Second)
 }
 
+func AddProxyConfigToCluster(oc oc.OcConfig, proxy *network.ProxyConfig) error {
+	cmdArgs := []string{"patch", "proxy", "cluster", "-p",
+		fmt.Sprintf(`{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s"}}`, proxy.HttpProxy, proxy.HttpsProxy, proxy.NoProxy),
+		"-n", "openshift-config", "--type", "merge"}
+
+	if err := waitForOpenshiftAPIServer(oc); err != nil {
+		return err
+	}
+	_, stderr, err := oc.RunOcCommand(cmdArgs...)
+	if err != nil {
+		return fmt.Errorf("Failed to add proxy details %v: %s", err, stderr)
+	}
+	return nil
+}
+
+func AddProxyToKubeletAndCriO(sshRunner *ssh.SSHRunner, proxy *network.ProxyConfig) error {
+	proxyTemplate := `[Service]
+Environment=HTTP_PROXY=%s
+Environment=HTTPS_PROXY=%s
+Environment=NO_PROXY=%s`
+	defaultNoProxyForCluster := ".cluster.local,.svc,10.128.0.0/14,172.30.0.0/16"
```

Comment:
Is this documented somewhere? Maybe it's better to put this directly in the template?

Reply:
> Is this documented somewhere? 

This you can get once the proxy is applied and check the status of it, this does take some time to come-up since some of the operator restart due to the change but below is what it looks like.

`10.128.0.0/14` is the pod network and `172.30.0.0/16` is service network.

```
$ oc get proxy -oyaml
[...]
  status:
    httpProxy: http://<my_proxy>
    httpsProxy: http://<my_proxy>
    noProxy: .cluster.local,.svc,10.128.0.0/14,127.0.0.1,172.30.0.0/16,api-int.crc.testing,api.crc.testing,etcd-0.crc.testing,localhost
[...]
```

> Maybe it's better to put this directly in the template?
@cfergeau sure we can put that in the template.

Reply:
I don't understand that link that you quote `noProxy: .cluster.local,.svc,10.128.0.0/14,127.0.0.1,172.30.0.0/16,api-int.crc.testing,api.crc.testing,etcd-0.crc.testing,localhost`
Is it you during your testing which specified `api-int.crc.testing`, `etcd-0.crc.testing`? or was it added automatically by some operator?

By "documented", I meant "is there some openshift documentation saying we should do that?" ?

Reply:
@cfergeau there is no doc from openshift side about it, this is taken care by the proxy operator and added automatically but since this `status` doesn't visible just after making the change in the proxy and need bit time so I used these value as constant because those are the one needed for `kubelet` and `crio` both